### PR TITLE
Add support for a dartdoc_options.yaml.

### PR DESCRIPTION
**Pull request by jcollins-g**
Mostly scaffolding and basic loading tests for a dartdoc options file.  This doesn't port any existing options over, instead creating only one to support #1610.  Porting command line options over and making the dartdoc_options file interact with command line options will be done later.

### CODE REVIEW

File: lib/src/dartdoc_options.dart [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+library dartdoc.dartdoc_options;
+
+import 'dart:io';
+
+import 'package:path/path.dart' as p;
+import 'package:yaml/yaml.dart';
+
+import 'logging.dart';
+
+Map<String, DartdocOptions> _dartdocOptionsCache = {};
```
Comment (devoncarew) [inline lib/src/dartdoc_options.dart]: `final`?

File: lib/src/dartdoc_options.dart [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+/// semantics to that of analysis_options.yaml,
+/// [documented here](https://www.dartlang.org/guides/language/analysis-options).
+/// It searches parent directories until it finds an analysis_options.yaml file,
+/// and uses built-in defaults if one is not found.
+///
+library dartdoc.dartdoc_options;
+
+import 'dart:io';
+
+import 'package:path/path.dart' as p;
```
Comment (devoncarew) [inline lib/src/dartdoc_options.dart]: I personally avoid single letter `as` imports

File: lib/src/dartdoc_options.dart (hunk 1 above)
Comment (jcollins-g) [reply lib/src/dartdoc_options.dart]: done

File: lib/src/dartdoc_options.dart (hunk 2 above)
Comment (jcollins-g) [reply lib/src/dartdoc_options.dart]: Yes, they're not great.  I just obliterated them all, replacing the inconsistent imports of path as path/p/etc with "pathLib".

**Pull request merged**